amqp/src/main/scala/akka/stream/alpakka/amqp/AmqpConnectionProvider.scala

```diff
@@ -199,6 +210,46 @@ object AmqpCredentials {
     AmqpCredentials(username, password)
 }
 
+final class AmqpSSLConfiguration(val protocol: Option[String] = None,
```

Comment:
Make the constructor private and add `apply` and `create` methods taking either `protocol` or `context` to the companion instead, please.